### PR TITLE
Followup for #12564 added transaction to param in model::find() and model::findFirst() 

### PR DESCRIPTION
**Pull request by chilimatic**
Hello!

* Type: bug fix | new feature | code quality | documentation
* Link to issue: #12409, #12564

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)?
- [x] I have checked that another pull request for this purpose does not exist.
- [ ] I wrote some tests for this PR.

Small description of change:

This change is to achieve the following

In Unit-testing one usual procedure is to create a transaction for a TestCase and after the execution rollback this change. 

Which means there is a "long transaction" going on and we want to check if the transaction actually is writing the correct values in the database. But the static context of model::find / model::findFirst does not get the transaction passed.

To achieve that passing a new index "transaction" containing an transaction in the $param array seams to be a viable solution. 

Another use-case can be having multiple transactions and wanting to look inside. 

Just to be precise -> ofc we could just get the connection and query manually inside of it, but I want to be as close to the framework as possible when testing certain services hence I want to use the model. 

I took your reviews and advice from the last attempt and used the params to not break the interfaces / change method signatures.  

the logic is implemented as follows: 

if you pass in a transaction via param in the find/findFirst method. It will set the transaction to query object and this transaction will be used for all queries and executions within this query context.

This should not affect any standard behaviour but still allow me to finally achieve cleaner unittests for my usecase.

I still owe you some unitests, but please look through it first so it's not after several hours of refactoring and cleanups that this gets rejected.

### CODE REVIEW

**Comment by chilimatic** [comment]
@sergeyklay any feedback on the functional code implementation ? if it looks okay to you we can move the semantics and syntaxation discussion and I will start writing tests for it.

File: phalcon/mvc/model/query.zep [hunk 1]
```diff
@@ -2544,21 +2584,14 @@ class Query implements QueryInterface, InjectionAwareInterface
...
-			} else {
-				let connection = model->getReadConnection();
-			}
+			let connection = this->getReadConnection(model, intermediate, bindParams, bindTypes);
 
-			// More than one type of connection is not allowed
-			let connectionTypes[connection->getType()] = true;
-			if count(connectionTypes) == 2 {
-				throw new Exception("Cannot use models of different database systems in the same query");
+			if (connection !== null) {
```
Comment (sergeyklay) [inline phalcon/mvc/model/query.zep]:
```zep
if typeof connection == "object"
```

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by chilimatic** [comment]
I did a force push to add the ticket reference correctly to all the commits

**Review by sergeyklay — APPROVED** [review]
LGTM

**Comment by chilimatic** [comment]
@sergeyklay thx,  I will write the unitests now :)

**Comment by sergeyklay** [comment]
@chilimatic Could you please use 3.3.x branch for that

**Comment by chilimatic** [comment]
@sergeyklay ofc, rebase my changes onto 3.3.x ?

**Comment by sergeyklay** [comment]
The 3.3.x branch is upstream for now

**Comment by chilimatic** [comment]
I will try to finish this asap. I just have some project deadlines comming up

**Comment by chilimatic** [comment]
added new issue #13226  and PR #13227

**Pull request closed without merging**